accepted/0003-interactive-audit-resolver.md

```diff
@@ -0,0 +1,92 @@
+# Interactive audit resolver
+
+## Summary
+
+This proposal is for adding means for a human to resolve issues from `npm audit` and interactively make decisions about each issue. Available as `npm audit resolve` command.
+
+Features:
+- Remember the resolutions and allow them to affect `npm audit` behavior.
+- Decision options include 'ignore', 'remind later', 'fix', 'remove'.
+- Allow tracking who resolved an issue and when using git history.
+
+
+## Motivation
+
+At times, running `npm audit fix` won't fix all audit issues. Fixes might not be available or the user might not want to apply some of them, eg. major version updates or build/test dependencies.
+
+It should be possible to make `npm audit` a step in a CI pipeline to go red every time a new vulnerability affects the project. Managing security of dependencies should be quick to update, effective, easy to audit over time and secure in itself.
+
+For that, the user needs a new tool which makes addressing issues one by one convenient even across a whole ecosystem of projects.
+
+## Detailed Explanation
+
+`npm audit resolve` runs audit and iterates over all actions from the output with a prompt for a decision on each. 
+If fix is available, it's offered as one of the options to proceed. Other options include: ignore, remind later and delete.
+
+All decisions are stored in `audit-resolv.json` file as key-value, where key is `${dependency path}|${advisory id}` and value is a structure specific to the resolution.
+
+`npm audit` reads `audit-resolv.json` and respects the resolution (ignores ignored issues, ignores issues postponed to a date in the future)
+
+### resolutions
+- **fix** runs the fix proposed in audit action and marks the issue as fixed in `audit-resolv.json`. On each subsequent run of `npm audit resolve` if the issue comes up again, the user is also warned that the problem was supposed to be fixed already.
+- **ignore** marks the issue as ignored and `npm audit` will no longer fail because of it (but should display a single line warning about the issue being ignored). If a new vulnerability is found in the same dependency, it does not get ignored. If another dependency is installed with the same vulnerability advisory id it is not ignored. If the same package is installed as a dependency to another dependency (different path) it is not ignored.
+- **postpone** marks the issue as postponed to a timestamp 24h in the future. Instead of ignoring an issue permanently just to make a build pass, one can postpone it when in rush, but make it show up as a CI faiure on the next working day. 
```

Comment:
The next working day is often more than 24 hours later; is this just meant as an example?

Reply:
Yes. I didn't intend it to count in working days. 

*If noone's at work and the build turns red, does it make a sound?*